### ISSUE TITLE
No error on assigning to literals

### ISSUE THREAD

**Issue by mgold** [elm/compiler#483]
All of these compile with no error:

``` haskell
3 = "wtf"
[3] = "wtf"
(3,4) = "wtf"
"wtf" = 3
```

The generated JS is `var $ = "wtf";` or whatever left-side literal was supplied.

Assigning to `[]` throws a runtime error and generates malformed JS, but it should be caught by the type checker under the same fix that will catch the above.

**Comment by TheSeamau5** [comment]
Bug still exists, this still compiles. 
```elm
import Html

x = 
  let 1 = 3
  in 1

main = Html.text (toString x)

```

It doesn't look harmful, but it should give an error

**Comment by Warry** [comment]
adding symbols to the function name also works (as it's expecting it to be an operator declaration)

```elm
w-tf="wtf"
```

**Comment by Warry** [comment]
also: https://github.com/elm-lang/elm-compiler/issues/770

**Comment by mgold** [comment]
Bug still exists in 0.15.1. It's kind of a weird issue in that it's obviously a bug, but it's not code that anyone but a QA engineer would write.

**Comment by Apanatshka** [comment]
The original issue may be resolved, since the compiler now gives an error, but I'd say that this "partial pattern" error deserves a place in the error message catalog the way it is now. There can be a much clearer error message for this specific situation, something like "The let binding is not introducing any new names, perhaps you want a case-of construct?".

**Comment by mgold** [comment]
Is `[] = "wtf"` also caught?

I agree with Jeff that the error could be clearer, but the fact that it's caught moves this from bug to error message catalog. Jeff would you like to open that issue?

**Comment by Apanatshka** [comment]
done: https://github.com/elm-lang/error-message-catalog/issues/52

I'm wondering if there are even that many patterns you can use in a `let` that don't result in a "partial pattern" error? I suppose tuples and records are still ok, but anything else?

**Comment by jvoigtlaender** [comment]
Well: the single constructor of a union type that has only one constructor.
